.buildkite/scripts/install_tools.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+source .buildkite/scripts/common.sh
+
+set -euo pipefail
+
+echo "--- Env preparation"
+
+if [ "${platform_type}" == "Linux" ]; then
+  DEBIAN_FRONTEND="noninteractive"
+  sudo mkdir -p /etc/needrestart
+  echo "\$nrconf{restart} = 'a';" | sudo tee -a /etc/needrestart/needrestart.conf > /dev/null
```

Comment:
I cannot understand this. Can you add a comment with the rationale and consequences if not done?

Reply:
added,
also, it can't be executed on the pre-command hook because of [this](https://buildkite.com/elastic/beats-metricbeat/builds/1753#018d5fa9-a67e-4026-844f-332f4f866c90/137) and [this](https://buildkite.com/elastic/beats-metricbeat/builds/1748#018d5f3b-0952-436b-9393-436d14122c0c/122-152)
@v1v

Reply:
That's why we can't use pre-command and we have to use separate scripts until we have our own images.

Reply:
I've been confirmed that sudo is not required in the `pre-command` hook, see the below answer:

> yes, it looks like pre-command runs as root (tested by creating .buildkite/hooks/pre-command with whoami)

Can you try again without sudo please?

Reply:
Sure